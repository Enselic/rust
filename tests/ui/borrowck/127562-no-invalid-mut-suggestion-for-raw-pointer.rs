// Regression test for https://github.com/rust-lang/rust/issues/127562

fn main() {
    let val = 2;
    let ptr = &raw const val;
    unsafe { *ptr = 3; }
}
