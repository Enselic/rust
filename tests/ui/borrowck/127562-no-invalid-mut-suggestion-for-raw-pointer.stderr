error[E0594]: cannot assign to `*ptr`, which is behind a `*const` pointer
  --> $DIR/127562-no-invalid-mut-suggestion-for-raw-pointer.rs:6:14
   |
LL |     unsafe { *ptr = 3; }
   |              ^^^^^^^^ `ptr` is a `*const` pointer, so the data it refers to cannot be written
   |
help: consider nordh changing this to be a mutable pointer
   |
LL |     let ptr = &mut raw const val;
   |                +++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0594`.
